src/revlanguage/workspace/RevVariable.cpp

```diff
@@ -210,41 +181,33 @@ size_t RevVariable::getReferenceCount(void) const
 RevObject& RevVariable::getRevObject(void) const
 {
     
-    if ( is_reference_var == true )
+    if ( isReferenceVariable() )
     {
         return referenced_variable->getRevObject();
     }
     
-    if ( is_vector_var == true && needs_building == true )
+    if ( isVectorVariable() && needs_building )
     {
-        needs_building = false;
-        
-        //        const std::set<int>& indices = the_var->getElementIndices();
-        //        if ( indices.empty() )
-        //        {
-        //            throw RbException("Cannot create a vector variable with name '" + identifier + "' because it doesn't have elements.");
-        //        }
-        
-        // @TODO: We actually might need a different workspace here than the user workspace. (Sebastian)
-        Environment &env = Workspace::userWorkspace();
-        
-        size_t max_index = getMaxElementIndex();
         std::vector<Argument> args;
-        for (size_t i = 1; i <= max_index; ++i)
+        int i=1;
+        for (auto& element_var: *vector_var_elements)
         {
-            std::string element_identifier = name + "[" + std::to_string(i) + "]";
-            RevPtr<RevVariable>& element_var = env.getVariable( element_identifier );
             // check that the element is not NULL
             if ( element_var == NULL || element_var->getRevObject() == RevNullObject::getInstance() )
```

Comment:
Given how `vector_var_elements` now works, will there ever be cases where `element_var` is `NULL`?

Reply:
I just checked, and this happens if you define `x[4]` but not `x[3]`.

Reply:
OK, I added a comment.